CommonCUDA.cmake

```diff
@@ -100,4 +104,26 @@ function(common_cuda_compile_options)
   endif()
 endfunction()
 
-
+# find CUDA capable device
+function(find_cuda_device)
+  file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/hasCudaDevice.cpp
+    "#include <cuda_runtime.h>
+     int main()
+     {
+         int deviceCount = 0;
+         if( cudaGetDeviceCount( &deviceCount ) != cudaSuccess )
+             deviceCount = 0;
+         return deviceCount > 0 ? 0 : 1;
+     }")
+
+  try_run(RUN_RESULT_VAR COMPILE_RESULT_VAR
+          ${CMAKE_CURRENT_BINARY_DIR}/hasCudaDevice
+          SOURCES ${CMAKE_CURRENT_BINARY_DIR}/hasCudaDevice.cpp
+          CMAKE_FLAGS -DINCLUDE_DIRECTORIES=${CUDA_INCLUDE_DIRS}
+                      -DLINK_LIBRARIES=${CUDA_LIBRARIES}
+          COMPILE_OUTPUT_VARIABLE COMPILE_OUTPUT_VAR
+          RUN_OUTPUT_VARIABLE RUN_OUTPUT_VAR)
+  if(COMPILE_RESULT_VAR AND NOT RUN_RESULT_VAR)
+    set(CUDA_DEVICE_FOUND TRUE CACHE BOOL "Whether a CUDA-capable device is present")
```

Comment:
I thought it could be convenient if several projects run it in a same configuration step (stable?). I don't really need it, you are the CMake experts, so I'll change it if you think it hurts more than anything.